src/game_api/script/lua_backend.cpp

```diff
@@ -36,27 +36,31 @@
 #include "usertypes/vanilla_render_lua.hpp" // for VanillaRenderContext
 #include "window_api.hpp"                   // for get_window
 
-std::recursive_mutex g_all_backends_mutex;
 std::vector<std::unique_ptr<LuaBackend::ProtectedBackend>> g_all_backends;
 std::unordered_map<int, HotKey> g_hotkeys;
 int g_hotkey_count = 0;
 
 LuaBackend::LuaBackend(SoundManager* sound_mgr, LuaConsole* con)
     : lua{get_lua_vm(sound_mgr), sol::create}, vm{acquire_lua_vm(sound_mgr)}, sound_manager{sound_mgr}, console{con}
 {
-    g_state = State::get().ptr_main();
+    g_state = State::get().ptr_local();
+    if (g_state == nullptr)
+    {
+        g_state = State::get().ptr_main();
+    }
+    ScriptState& state = local_state_datas[g_state].state;
```

Comment:
Isn't backend constructed ASAP in playlunky? then this will always be main state anyway

Reply:
not sure about when they are constructed, are they when you create a new script in OL?

Reply:
Was thinking more about PL

Reply:
Yeah but this code is for both PL and OL, also handling what happens in OL makes sense to me